filebeat/autodiscover/builder/hints/config.go

```diff
@@ -53,7 +51,16 @@ func (c *config) Unpack(from *common.Config) error {
 	}
 
 	if config, err := from.Child("default_config", -1); err == nil {
-		c.DefaultConfig = config
+		fields := config.GetFields()
+		if len(fields) == 1 && fields[0] == "enabled" {
+			// only enabling/disabling default config:
+			if err := c.DefaultConfig.Merge(config); err != nil {
+				return nil
+			}
+		} else {
+			// full config provided, discard default
+			c.DefaultConfig = config
+		}
```

Comment:
What if a full config is provided, but also with the enabled setting set to true?
Would this work?
```
          hints.default_config:
            type: container
            enabled: true
            paths:
              - "/var/log/pods/${data.kubernetes.pod.uid}/${data.kubernetes.container.name}/*.log"
```

Reply:
It should, as I'm checking number of fields == 1

Reply:
Oh, right